src/main/java/it/gov/pagopa/pu/send/service/UploadService.java

```diff
@@ -0,0 +1,10 @@
+package it.gov.pagopa.pu.send.service;
+
+import it.gov.pagopa.pu.send.dto.DocumentDTO;
+import java.io.IOException;
+import java.security.NoSuchAlgorithmException;
+import java.util.Optional;
+
+public interface UploadService {
+  Optional<String> uploadFileToS3(String sendNotidicationId, DocumentDTO documentDTO) throws IOException, NoSuchAlgorithmException;
```

Comment:
Let's not expose info about the integration
```suggestion
  Optional<String> upload(String sendNotidicationId, DocumentDTO documentDTO);
```